pycaret/containers/models/time_series.py

```diff
@@ -910,6 +920,69 @@ def _set_tune_distributions(self) -> dict:
         return {}
 
 
+class ProphetContainer(TimeSeriesContainer):
+    def __init__(self, globals_dict: dict) -> None:
+        logger = get_logger()
+        np.random.seed(globals_dict["seed"])
+        self.gpu_imported = False
+
+        try:
+            from sktime.forecasting.fbprophet import Prophet  # type: ignore
+        except ImportError:
+            logger.warning("Couldn't import sktime.forecasting.fbprophet")
+            self.active = False
+            return
+
+        sp = globals_dict.get("seasonal_period")
+        self.sp = sp if sp is not None else 1
+
+        self.seasonality_present = globals_dict.get("seasonality_present")
+        self.freq = globals_dict.get("freq")
+
+        args = self._set_args
+        tune_args = self._set_tune_args
+        tune_grid = self._set_tune_grid
+        tune_distributions = self._set_tune_distributions
+
+        leftover_parameters_to_categorical_distributions(tune_grid, tune_distributions)
+
+        super().__init__(
+            id="prophet",
+            name="Prophet",
+            class_def=Prophet,
+            args=args,
+            tune_grid=tune_grid,
+            tune_distribution=tune_distributions,
+            tune_args=tune_args,
+            is_gpu_enabled=self.gpu_imported,
+            is_turbo=False,
+        )
+
+    @property
+    def _set_args(self) -> dict:
+        return {} 
+
+    @property
+    def _set_tune_args(self) -> dict:
+        return {}
+
+    @property
+    def _set_tune_grid(self) -> dict:
```

Comment:
[Check here](https://facebook.github.io/prophet/docs/diagnostics.html#hyperparameter-tuning) on what parameters need to be tuned and how. Some of these could be added to the `tune_distribution` with log sampling.

Reply:
makes sense, thanks for the reference.

Reply:
@ngupta23 added the tuning according to the reference. But, there's an issue with cloning the `Prophet` estimator after finding the best parameters, I've opened an issue explaining the details [link](https://github.com/pycaret/pycaret/issues/1311)